lib/querystring.js

```diff
@@ -194,6 +195,9 @@ function stringify(obj, sep, eq, options) {
     (encode === qsEscape ? encodeStringified : encodeStringifiedCustom);
 
   if (obj !== null && typeof obj === 'object') {
+    obj = obj instanceof Map ? [...obj.entries()].reduce(
+      (acc, [key, value]) => (acc[key] = value, acc), {}) : obj;
+
```

Comment:
This works but it's not the fastest way of iterating through these. Given that it's going to be an uncommon code path I'm fine with that tho